src/integrationTest/java/uk/gov/hmcts/reform/civil/controllers/CallbackControllerTest.java

```diff
@@ -11,16 +12,109 @@
+    }
 
+    @Test
+    @SneakyThrows
+    void shouldReturnNotFoundWhenCallbackHandlerIsNotImplemented() {
         doPost(BEARER_TOKEN, callbackRequest, CALLBACK_URL, "invalid-callback-type")
             .andExpect(status().isNotFound());
     }
+
+    @Test
+    @SneakyThrows
+    void shouldReturnOkWithValidCallbackType() {
+        doPost(BEARER_TOKEN, callbackRequest, CALLBACK_URL, VALID_CALLBACK_TYPE2)
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    @SneakyThrows
+    void shouldReturnOkWithValidCallbackTypeAndPageId() {
+        doPost(BEARER_TOKEN, callbackRequest, CALLBACK_PAGE_ID_URL, VALID_CALLBACK_TYPE, VALID_PAGE_ID)
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    @SneakyThrows
+    void shouldReturnNotFoundWithInvalidCallbackTypeButValidPage() {
+        doPost(BEARER_TOKEN, callbackRequest, CALLBACK_PAGE_ID_URL, INVALID_CALLBACK_TYPE, VALID_PAGE_ID)
+            .andExpect(status().isNotFound());
+    }
+
+    @Test
+    @SneakyThrows
+    void shouldReturnNotFoundWithValidCallbackTypeButInvalidPage() {
+        doPost(BEARER_TOKEN, callbackRequest, CALLBACK_PAGE_ID_URL, VALID_CALLBACK_TYPE, INVALID_PAGE_ID)
+            .andExpect(status().isNotFound());
+    }
+
+    @Test
+    @SneakyThrows
+    void shouldReturnNotFoundWithInvalidCallbackTypeAndPage() {
+        doPost(BEARER_TOKEN, callbackRequest, CALLBACK_PAGE_ID_URL, INVALID_CALLBACK_TYPE, INVALID_PAGE_ID)
+            .andExpect(status().isNotFound());
+    }
+
+    @Test
+    @SneakyThrows
+    void shouldReturnOkWithValidCallbackTypeAndVersion() {
+        doPost(BEARER_TOKEN, callbackRequest, CALLBACK_VERSION_URL, V1, VALID_CALLBACK_TYPE)
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    @SneakyThrows
+    void shouldReturnBadRequestWithInvalidVersion() {
+        doPost(BEARER_TOKEN, callbackRequest, CALLBACK_VERSION_URL, INVALID_VERSION, VALID_CALLBACK_TYPE)
+            .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    @SneakyThrows
+    void shouldReturnNotFoundWithValidVersionAndInvalidType() {
+        doPost(BEARER_TOKEN, callbackRequest, CALLBACK_VERSION_URL, V1, INVALID_CALLBACK_TYPE)
+            .andExpect(status().isNotFound());
+    }
+
+    @Test
+    @SneakyThrows
+    void shouldReturnOkWithAllValidParams() {
+        doPost(BEARER_TOKEN, callbackRequest, CALLBACK_VERSION_PAGE_ID_URL, V1, VALID_CALLBACK_TYPE, VALID_PAGE_ID)
+            .andExpect(status().isOk());
+    }
+
+    @Test
+    @SneakyThrows
+    void shouldReturnBadRequestWithInvalidVersionOnly() {
+        doPost(BEARER_TOKEN, callbackRequest, CALLBACK_VERSION_PAGE_ID_URL, INVALID_VERSION, VALID_CALLBACK_TYPE, VALID_PAGE_ID)
+            .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    @SneakyThrows
+    void shouldReturnBadRequestWithNullBody() {
+        doPost(BEARER_TOKEN, null, CALLBACK_PAGE_ID_URL, VALID_CALLBACK_TYPE, VALID_PAGE_ID)
+            .andExpect(status().isBadRequest());
+    }
```

Comment:
Seems to me you have 3 categories of tests here: those expecting `status().isOk()`, those expecting `status().isNotFound()` and those expecting `status().isBadRequest()`, working on different combinations of URLs and arguments.

Consider using `@ParameterizedTest`. It may reduce the amount of duplicated code significantly.